server/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java

```diff
@@ -135,23 +136,27 @@ private StoreFilesMetaData listStoreMetaData(ShardId shardId) throws IOException
                         return new StoreFilesMetaData(shardId, Store.MetadataSnapshot.EMPTY, Collections.emptyList());
                     }
                 }
+                if (request.getCustomDataPath() != null) {
```

Comment:
NIT: maybe do something like this:

```diff
diff --git a/server/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java b/server/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java
index a7874ab2984..48edbf37628 100644
--- a/server/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java
+++ b/server/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java
@@ -119,7 +119,6 @@ public class TransportNodesListShardStoreMetaData extends TransportNodesAction<T
         boolean exists = false;
         try {
             IndexService indexService = indicesService.indexService(shardId.getIndex());
-            final String customDataPath;
             if (indexService != null) {
                 IndexShard indexShard = indexService.getShardOrNull(shardId.id());
                 if (indexShard != null) {
@@ -136,17 +135,15 @@ public class TransportNodesListShardStoreMetaData extends TransportNodesAction<T
                         return new StoreFilesMetaData(shardId, Store.MetadataSnapshot.EMPTY, Collections.emptyList());
                     }
                 }
-                if (request.getCustomDataPath() != null) {
-                    customDataPath = request.getCustomDataPath();
-                } else {
-                    // TODO: Fallback for BWC with older ES versions. Remove this once request.getCustomDataPath() always returns non-null
-                    customDataPath = indexService.getIndexSettings().customDataPath();
-                }
+            }
+            final String customDataPath;
+            if (request.getCustomDataPath() != null) {
+                customDataPath = request.getCustomDataPath();
             } else {
-                if (request.getCustomDataPath() != null) {
-                    customDataPath = request.getCustomDataPath();
+                // TODO: Fallback for BWC with older ES versions. Remove this once request.getCustomDataPath() always returns non-null
+                if (indexService != null) {
+                    customDataPath = indexService.getIndexSettings().customDataPath();
                 } else {
-                    // TODO: Fallback for BWC with older ES versions. Remove this once request.getCustomDataPath() always returns non-null
                     IndexMetaData metaData = clusterService.state().metaData().index(shardId.getIndex());
                     if (metaData != null) {
                         customDataPath = new IndexSettings(metaData, settings).customDataPath();
```

to keep it shorter and the BwC logic in a single place?

Reply:
good idea, addressed in 3bd2843979a